engine/engine/content/builtins/scripts/edn.lua

```diff
@@ -68,38 +55,34 @@ local function encode_number(val)
 end
 
 local function encode_string(val)
-  return '"' .. val:gsub('[%z\1-\31\\"]', escape_char) .. '"'
+  return '"'..val:gsub('[%z\1-\31\\"]', escape_char)..'"'
```

Comment:
Do we ever call encode_string() on the same string multiple times? If so, perhaps cache the result and trade memory for performance?

Reply:
Great Idea!
With strings cache:
```
DEBUG:SCRIPT: Result time = 	5.7142062187195	 cached_string	2056917	 encoded_strs	476642
-- and if I don't clean cache (which make sence because I can just step over) it is:
DEBUG:SCRIPT: Result time = 	5.2303631305695	 cached_string	2533328	 encoded_strs	68
DEBUG:SCRIPT: Result time = 	5.1561901569366	 cached_string	2533404	 encoded_strs	40
DEBUG:SCRIPT: Result time = 	5.0541839599609	 cached_string	2533514	 encoded_strs	60
DEBUG:SCRIPT: Result time = 	5.0240840911865	 cached_string	2533549	 encoded_strs	34
```
I'll add some limit when I'll clean cache, but it works good.
Also it gives food for thought about too much work we does all the time

Reply:
I added strings cache